extra_scripts/nrf52_lto.py

```
@@ -199,7 +209,85 @@ def _assert_isr_handlers_survived(source, target, env):
+                    )
+
+        def _kinds(nm_out):
+            kinds = {}
+            for line in nm_out.split("\n"):
+                f = line.split()
+                if len(f) >= 3:
+                    kinds[f[-1]] = f[-2]
+            return kinds
+
+        elf = env.subst("$BUILD_DIR/${PROGNAME}.elf")
+        elf_kind = _kinds(subprocess.check_output([_NM, elf], universal_newlines=True))
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)